src/index.js

```diff
@@ -0,0 +1,172 @@
+import UploaderService from './service';
+
+const KILOBYTE = 1000,
+    MEGABYTE = 1024;
+
+class UploadController {
+
+    constructor({url, attachments = {}, maxFiles = 10, headers = {}, optionsObject = {}, updateCb, onLoad, maxFilesText = ''}) {
+        this.attachments = attachments;
+        this.maxFiles = maxFiles;
+        this.url = url;
+        this.optionsObject = optionsObject;
+        this.headers = headers;
+        this.fileCounter = Object.keys(this.attachments).length;
+        this.updateCb = updateCb;
+        this.onLoad = onLoad;
+        this.maxFilesText = maxFilesText;
+
+        this.onAttach = this.onAttach.bind(this);
+        this.onAbort = this.onAbort.bind(this);
+    }
+
+    static itemId(name) {
+        return `file_${name.match(/[a-zA-Z0-9_]/gi).join('')}`;
+    }
+
+    static maxFilesReached(fileCount, maxFiles) {
+        return fileCount >= maxFiles;
+    }
+
+    static stripNameFromExtension(name) {
+
+        const lastDot = name.lastIndexOf('.'),
+            nameLength = name.length,
+            hasExtension = (nameLength - lastDot) <= 5;
```

Comment:
The hardest part is determining whether we've found an extension or a period in the filename.